src/boot/pprint.ml

```diff
@@ -1,244 +1,492 @@
+      !ref_indent
+      print_tm (MATCH, t)
+      p
+      print_tm (IF, then_)
+      print_tm (IF, else_)
 
-(* TODO: Print mlang part as well*)
-and pprintML tml =
+  | TmUse(_,l,t) ->
+    let l = string_of_ustring l in
+    fprintf fmt "@[<hov 0>use %s in@ %a@]"
+      l print_tm (MATCH, t)
+
+  | TmUtest(_,t1,t2,t3) ->
+    fprintf fmt "@[<hov 0>\
+                   @[<hov %d>\
+                     utest@ \
+                     @[<hov 0>\
+                       %a with@ \
+                       %a in\
+                     @]\
+                   @]\
+                   @ %a\
+                 @]"
+      !ref_indent
+      print_tm (MATCH, t1)
+      print_tm (MATCH, t2)
+      print_tm (MATCH, t3)
+
+  | TmClos(_,x,ty,t1,_) ->
+    let x = string_of_ustring x in
+    let ty = ty |> ustring_of_ty |> string_of_ustring in
+    fprintf fmt "@[<hov %d>clos %s:%s.@ %a@]"
+      !ref_indent x
+      ty
+      print_tm (LAM, t1)
+
+  | TmFix _ -> fprintf fmt "fix"
+
+(** Print an environment on the given formatter. *)
+and print_env fmt env =
+  let print i t = (fun fmt -> fprintf fmt "%d -> %a" i print_tm (MATCH, t)) in
+  let inner = List.mapi print env in
+  fprintf fmt "[@[<hov 0>%a@]]" concat (COMMA,inner)
+
+(** Helper function for configuring the string formatter and printing *)
+let str_formatter_print
+    ?(debruijn   = !enable_debug_debruijn_print)
+    ?(indent     = 2)
+    ?(max_indent = 68)
+    ?(margin     = 80)
+    ?(max_boxes  = max_int)
+    ?(prefix     = "")
+    printer arg =
+
+  (* Configure global settings *)
+  ref_debruijn := debruijn;
+  ref_indent   := indent;
+  pp_set_margin     str_formatter margin;
+  pp_set_max_indent str_formatter max_indent;
+  pp_set_max_boxes  str_formatter max_boxes;
+
+  (* Make sure formatter is cleared *)
+  ignore (flush_str_formatter ());
+
+  (* Print a prefix *)
+  fprintf str_formatter "%s" prefix;
+
+  (* Do the actual printing *)
+  printer str_formatter arg;
+
+  (* Return result string and clear formatter *)
+  flush_str_formatter ()
+
+(** Convert terms to strings.
+ *  TODO Messy with optional arguments passing. Alternatives? *)
+let string_of_tm ?debruijn ?indent ?max_indent ?margin ?max_boxes ?prefix t =
+  str_formatter_print ?debruijn ?indent ?max_indent ?margin ?max_boxes ?prefix
+    print_tm (MATCH, t)
+
+(** Converting constants to strings.
+ *  TODO Messy with optional arguments passing. Alternatives? *)
+let string_of_const ?debruijn ?indent ?max_indent ?margin ?max_boxes ?prefix c =
+  str_formatter_print ?debruijn ?indent ?max_indent ?margin ?max_boxes ?prefix
+    print_const c
+
+(** Converting environments to strings.
+ *  TODO Messy with optional arguments passing. Alternatives? *)
+let string_of_env ?debruijn ?indent ?max_indent ?margin ?max_boxes ?prefix e =
+  str_formatter_print ?debruijn ?indent ?max_indent ?margin ?max_boxes ?prefix
+    print_env e
+
+(** Old printing function for terms.
+ *  @deprecated Use string_of_tm instead. *)
```

Comment:
Question for meeting: do we want to deprecate and then remove, or just remove and make necessary changes?